Sources/MapboxCoreNavigation/Directions+NavigationProviderRequest.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+import MapboxDirections
+
+extension URLSessionDataTask: NavigationProviderRequest {
+    public var id: UInt64 {
+        UInt64(taskIdentifier)
+    }
+}
+
+extension Directions: NavigationProvider {
+    
+    @discardableResult public func requestRoutes(options: RouteOptions, completionHandler: @escaping RouteCompletionHandler) -> NavigationProviderRequest? {
+        return calculate(options, completionHandler: completionHandler)
+    }
+    
+    @discardableResult public func requestRoutes(options: MatchOptions, completionHandler: @escaping MatchCompletionHandler) -> NavigationProviderRequest? {
+        return calculate(options, completionHandler: completionHandler)
+    }
+    
+    @discardableResult public func refreshRoute(indexedRouteResponse: IndexedRouteResponse, fromLegAtIndex: UInt32, completionHandler: @escaping RouteCompletionHandler) -> NavigationProviderRequest? {
+        guard case let .route(routeOptions) = indexedRouteResponse.routeResponse.options else {
+            preconditionFailure("Invalid route data passed for refreshing.")
```

Comment:
This only happens if the options object is a MatchOptions, right? Let’s say something about that in the error message.